### DISCUSSION TITLE
It wont work by using curl or python script to call the request

### DISCUSSION THREAD

**Discussion by OriLiMu** [coddingtonbear/obsidian-local-rest-api#37] [Q&A]
# It is worked  when I use the web test. 
![image](https://user-images.githubusercontent.com/11295355/201507784-0e8a2f45-97b5-4982-8cf6-04ef04141486.png)

# But I get error when using curl in terminal.  Show nothing when use curl.  
http:
![image](https://user-images.githubusercontent.com/11295355/201507819-9fc436f8-ef67-43fb-ac32-8a9f961a7ab7.png)
https:  ssl error.   
![image](https://user-images.githubusercontent.com/11295355/201507831-42ac961b-585c-4a1c-961a-65c63ebbf2c1.png)

# Python script also wont work 

```python
def LocalRestApiTest():
    import requests
    url = 'http://127.0.0.1:27123/commands/'
    headers = {
        'accept': 'application/json',
        'Accept-Encoding': 'gzip, deflate',
        'Accept-Language': 'zh-CN,zh;q=0.9,en;q=0.8,en-US;q=0.7,ja;q=0.6,zh-TW;q=0.5',
        'Authorization': 'Bearer xxx',
        'Connection': 'keep-alive',
        'Content-Type': 'application/json; charset=utf-8',
        'Host': '127.0.0.1:27123',
    }
    response = requests.get(url, headers=headers, verify=False)
    print('test')

```

Response is 502. 
![image](https://user-images.githubusercontent.com/11295355/201507870-d370a335-fe46-4ae3-bd3a-266a4f55c05a.png)

 So why this happened and how to solve it?

**Comment by coddingtonbear** [comment]
Just to start, you shold note that the HTTPS server runs on a different port from the HTTP.  Your `curl` executions above appear to use the HTTP port for both HTTP and HTTPS connections; your strange "unexpected eof" error is explained by that.  Even if you fix that, though, I'd expect that you'd continue to get a 502 error.

What I'd recommend, probably for your first step is to open the inspector in obsidian `Ctrl+Shift+I` (or `Opt+Cmd+I` if you're on a mac) and seeing if you an error is displayed in the console when you run your request.
